manual/src/tracing.md

```diff
@@ -0,0 +1,127 @@
+# Tracing and Logging
+
+The driver utilizes the [`tracing`](https://crates.io/crates/tracing) crate to emit events at points of interest. To enable this, you must turn on the `tracing-unstable` feature flag.
+
+## Stability Guarantees
+This functionality is considered unstable as the `tracing` crate has not reached 1.0 yet. Future minor versions of the driver may upgrade the `tracing` dependency
+to a new version which is not backwards-compatible with `Subscriber`s that depend on older versions of `tracing`.
+Additionally, future minor releases may make changes such as:
+* add or remove tracing events
+* add or remove values attached to tracing events
+* change the types and/or names of values attached to tracing events
+* add or remove driver-defined tracing spans
+* change the severity level of tracing events
+
+Such changes will be called out in release notes.
+
+## Event Targets 
+
+Currently, events are emitted under the following targets:
+
+| Target                      | Description                                                                                                   |
+|-----------------------------|---------------------------------------------------------------------------------------------------------------|
+| `mongodb::command`          | Events describing commands sent to the database and their success or failure.                                 |
+| `mongodb::server_selection` | Events describing the driver's process of selecting a server in the database deployment to send a command to. |
+| `mongodb::connection`       | Events describing the behavior of driver connection pools and the connections they contain.                   |
+
+## Consuming Events
+To consume events in your application, in addition to enabling the `tracing-unstable` feature flag, you must either register a `tracing`-compatible subscriber or a `log`-compatible logger, as detailed in the following sections.
+
+### Consuming Events with `tracing`
+
+To consume events with `tracing`, you will need to register a type implementing the `tracing::Subscriber` trait in your application, as [discussed in the `tracing` docs](https://docs.rs/tracing/latest/tracing/#in-executables).
+
+Here's a minimal example of a program using the driver which uses a tracing subscriber.
+
+First, add the following to `Cargo.toml`:
+```toml,no_run
+tracing = "LATEST_VERSION_HERE"
+tracing-subscriber = "LATEST_VERSION_HERE"
+mongodb = { version = "LATEST_VERSION_HERE", features = ["tracing-unstable"] }
+```
+
+And then in `main.rs`:
+
+```rust,no_run
+# extern crate mongodb;
+# extern crate tokio;
+# extern crate tracing_subscriber;
+# use std::env;
+# use mongodb::{Client, bson::doc, options::ClientOptions};
+
+#[tokio::main]
+async fn main() {
```

Comment:
Can we update this to return a `Result<()>` so that we can use `?` below rather than unwraps?